### PR TITLE
feat: switch from file cache to sql lite

### PR DESCRIPTION
**Pull request by DenhamPreen**

### CODE REVIEW

**Comment by DenhamPreen** [comment]
image

Flying through events again :)

File: package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "codegen": "envio codegen",
     "dev": "envio dev",
     "test": "pnpm mocha",
-    "start": "ts-node generated/src/Index.bs.js"
+    "start": "ts-node generated/src/Index.bs.js",
+    "migrate": "ts-node src/migrate.ts"
```
Comment (DenhamPreen) [inline package.json:13]: I left this script here but it should be deleted, just wanted to save the code in git incase

File: src/contracts/corev1.ts [hunk 2]
```diff
@@ -5,124 +5,141 @@ import { Cache, CacheCategory } from "../lib/cache";
...
-        cache.add({ [conditionId]: entry as any });
-
-        return entry;
-    } catch (err) {
-        console.error("An error occurred", err);
-        throw err; // or handle the error as needed
-    }
+  const conditionId = _conditionId.toString();
+  const cache = Cache.init(CacheCategory.ConditionV1, chainId);
+  const _condition = await cache.read(conditionId);
```
Comment (DenhamPreen) [inline src/contracts/corev1.ts:27]: this 'await' is the only difference here

File: src/lib/cache.ts [hunk 3]
```diff
@@ -60,93 +75,103 @@ export class Cache {
...
-      fs.writeFileSync(this.file, prepared);
-    } catch (error) {
-      console.error(error);
-    }
-  }
-
-  static resolve(key: string) {
-    return path.join(Entry.folder, key.toLowerCase().concat(".json"));
+    const id = keys[0];
+    // todo: this could be smarter and actually store the data in columns and not just a id data field
```
Comment (DenhamPreen) [inline src/lib/cache.ts:161]: This is a notable improvement but maybe not worth the encoding and decoding, it would allow for richer querying though.

**Review by AlexTheLion123 — APPROVED** [review]

**Pull request merged**